worker/caasapplicationprovisioner/doc.go

```diff
@@ -0,0 +1,12 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package caasapplicationprovisioner is a state reconcilliation
+// engine for caas. We create a sub-worker for each application
```

Comment:
Can you elaborate on what "sub-worker" means. Are you saying that there is a single "primary" `caasapplicationprovisioner` worker for the model, which manages a bunch of "subordinate" `caasapplicationprovisioner` workers, one per application in the model?